File: Tatarstan/Gis/Bars.Gkh.Gis/libs/B4/controller/AnalysisReport.js
Ext.define('B4.controller.AnalysisReport', {
    extend: 'B4.base.Controller',

    mixins: { context: 'B4.mixins.Context' },
    views: ['analysisreport.AnalysisReportForm'],
    stores: ['AnalysisReport'],
    mainView: 'analysisreport.AnalysisReportForm',
    mainViewSelector: 'analysisreportform',

    init: function () {
        var me = this;
        me.control({
            scope: me,
            'analysisreportform': {
                'show': me.onLoad
            }
        });

        me.callParent(arguments);
    },

    index: function (apartmentId, serviceId, supplierId, month, year) {
        var me = this,
            view = me.getMainView() || Ext.widget('analysisreportform');

        me.bindContext(view);

        me.setContextValue(view, 'apartmentId', apartmentId);
        me.setContextValue(view, 'serviceId', serviceId);
        me.setContextValue(view, 'supplierId', supplierId);
        me.setContextValue(view, 'month', month);
        me.setContextValue(view, 'year', year);

        me.application.deployView(view);
        view.show();
    },

    onLoad: function () {
        var me = this,
            win = me.getMainView();

        Ext.util.CSS.createStyleSheet(
            '.no-icon { margin-left: -5px; !important; ' 
        );

        var apartmentId = me.getContextValue(win, 'apartmentId');
        var serviceId = me.getContextValue(win, 'serviceId');
        var supplierId = me.getContextValue(win, 'supplierId');
        var month = me.getContextValue(win, 'month');
        var year = me.getContextValue(win, 'year');

        B4.Ajax.request({
            url: 'AnalysisReport/GetAnalysisReportData',
            method: 'POST',
            params: {
                apartmentId: apartmentId,
                serviceId: serviceId,
                supplierId: supplierId,
                month: month,
                year: year
            }
        }).next(function (jsonResp) {
            var resultData = Ext.JSON.decode(jsonResp.responseText).data;

            win.down('label[name=LS]').setText(resultData.ApartmentId);
            win.down('label[name=CalcMonth]').setText(resultData.CalcMonth);
            win.down('label[name=Address]').setText(resultData.Address);
            win.down('label[name=Supplier]').setText(resultData.Supplier);

            win.setTitle('Протокол расчета начислений по услуге \"' + resultData.Service + '\"');

            var reportTree = win.down('treepanel[name=reportTree]');
            var root = reportTree.getRootNode();

            me.addNode(resultData.ReportTree, root);

            root.expand();
        });
    },

    addNode: function (data, parentNode, operation) {
        var me = this;

        var text = data.Title + ' <b>' + data.Charge + '</b>';

        var node = parentNode.appendChild({
           Title: text,
           //iconCls: 'no-icon',
           Note: data.Note
        });

        if (operation != null) {
            parentNode.appendChild({
                Title: '<b>' + operation + '</b>',
                iconCls: 'no-icon',
                leaf: true
            });
        }

        if (data.Left != null) {
            me.addNode(data.Left, node, data.ChildsOperation);
        }

        if (data.Right != null) {
            me.addNode(data.Right, node);
        }
    }
});

File: Tatarstan/Gis/Bars.Gkh.Gis/Entities/AnalysisReport.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Bars.Gkh.Gis.Entities
{
    class AnalysisReport
    {
        /// <summary>
        /// Идентификатор квартиры
        /// </summary>
        public virtual long ApartmentId { get; set; }

        /// <summary>
        /// Расчетный месяц
        /// </summary>
        public virtual string CalcMonth { get; set; }

        /// <summary>
        /// Наименование услуги
        /// </summary>
        public virtual string Service { get; set; }

        /// <summary>
        /// Наименование поставщика
        /// </summary>
        public virtual string Supplier { get; set; }

        /// <summary>
        /// Адрес
        /// </summary>
        public virtual string Address { get; set; }

        /// <summary>
        /// Дерево расчетов
        /// </summary>
        public virtual AnalysisReportNode ReportTree { get; set; }

        ///// <summary>
        ///// Сумма начисления
        ///// </summary>
        //public virtual decimal Charge { get; set; }
        ///// <summary>
        ///// Тариф
        ///// </summary>
        //public virtual decimal Tariff { get; set; }
        ///// <summary>
        ///// Расход
        ///// </summary>
        //public virtual decimal Consumption { get; set; }


    }

    class AnalysisReportNode
    {
        /// <summary>
        /// Заголовок
        /// </summary>
        public virtual string Title { get; set; }

        /// <summary>
        /// Сумма начисления
        /// </summary>
        public virtual decimal Charge { get; set; }

        /// <summary>
        /// Примечание
        /// </summary>
        public virtual string Note { get; set; }


        /// <summary>
        /// Операция между потомками
        /// </summary>
        public virtual string ChildsOperation { get; set; }

        /// <summary>
        /// Левый потомок
        /// </summary>
        public virtual AnalysisReportNode Left { get; set; }

        /// <summary>
        /// Правый потомок
        /// </summary>
        public virtual AnalysisReportNode Right { get; set; }
        ///// <summary>
        ///// Единицы измерения
        ///// </summary>
        //public virtual string Measure { get; set; }
    }
}
